content/posts/2021/01/2021-11-04-Getting-Started-With-React/index.md

```diff
@@ -0,0 +1,334 @@
+
+Click [<span style="color:blue">**this link**</span>]( https://code.visualstudio.com/download ) to download the version that suits your platform.
+That's enough development set up for now.
+
+
+## Creating a React App
+
+The next step is now creating a React app. Thank God for the folks at Facebook who made it easy for us to do that.
+All we to do is run this command in our terminal.
+
+ > npx create-react-app my-app
+
+This craetes a React project called my-app.
+![Output on the terminal](./images/vscode.png)
+
+Realize we've been directed on how to start our app
+ > cd my-app
+ > yarn start
+
+
+ Run this commands on your terminal, this will start a development server and open a web browser for you.
+ ![Web browser](./images/react-app.png)
+
+ You;ve just set up your first React app, if you want to know more about what happens in the background visit this link [Create React App}(https://github.com/facebook/create-react-app)
+
+
+## Exploring Create React App
+
+![Here's the file structure](./images/react-file-structure.png)
+
+After running  ``npx create-react-app yourApp``, you will find your app root has three folders, ``node_modules/``, ``public/``, ``src/`` and three files ``.gitignore``, ``package.json``, ``README.md`` and ``yarn.lock``.
+
+### Running create-react-app
+
+Let's break down what you just did. The first line in the output is from ``npx`` . There are two things `npx` can do when you give it the command. It first looks in your local machine for the package to run it. If it doesn't find one, it starts looking for ``npm``'s package list. The first message in the output is just letting you know that it handled getting it.
+
+Next inside ``create-react-app``, your command runs through a series of checks. It finds information about your system, looks for any flags attached and whether you specified a folder to output. Then the actual ``createApp`` function runs. This checks if the directory you choose is a safe one. It does this by looking for log files, if it fails any of this checks you will receive an error message.
+
+Next it begins to form a ``package.json`` file with the default values ``name: yourAppName``, ``version: 0.1.0`` and ``private: true``. This is written to the directory you specified. Here it determines if it should use ``yarn`` or ``npm`` based on what you have in your machine. A series of checks are run on the version of ``npm/yarn``, ``node version`` to select the correct version of ``react-scripts`` to use.
+
+The next function is called ``run()`` . This continues the installation. Right now the only output you've seen so far is is the npx line  "Creating a react app in <YourDirectory>". The dependency list is created. By default this icludes ``react``, ``react-dom`` and the version of ``react-scripts`` either from the internet if you are online or the local cache if you are not.
```

Comment:
This line : 
```
Right now the only output you've seen so far is is the npx line  "Creating a react app in <YourDirectory>". The dependency list is created. By default this icludes ``react``, ``react-dom`` and the version of ``react-scripts`` either from the internet if you are online or the local cache if you are not. 
```

This causes the markdown parser to bug out on build. 

What might be causing this is `<YourDirectory>` 

Suggested fix: 
```diff
- The next function is called ``run()`` . This continues the installation. Right now the only output you've seen so far is is the npx line  "Creating a react app in <YourDirectory>". The dependency list is created. By default this icludes ``react``, ``react-dom`` and the version of ``react-scripts`` either from the internet if you are online or the local cache if you are not.
+ The next function is called ``run()`` . This continues the installation. Right now the only output you've seen so far is is the npx line  "Creating a react app in `<YourDirectory>`". The dependency list is created. By default this icludes ``react``, ``react-dom`` and the version of ``react-scripts`` either from the internet if you are online or the local cache if you are not. 
```

Notice the additional back-ticks `<YourDirectory>`